inc/cub3d_render.h

```diff
@@ -16,33 +16,188 @@
 # include "data_structure.h"
 # include <stddef.h>
 
+/**
+ * @brief Init the map translation and the backbuffer for rendering
+ * 
+ * @param[in,out] data data structure
+ * @return t_data* data if successful, NULL otherwise
+ */
 t_data	*cub_init(t_data *data);
+/**
+ * @brief Translate the 2d map extracted from input file into a vector of int
+ * 
+ * @param[in,out] data data structure
+ * @return int 1 OK, 0 otherwise
+ */
 int		cub_translate_map(t_data *data);
+/**
+ * @brief Init cub for rendering
+ * 
+ * @param[in,out] data data structure
+ * @return int 0 OK, error otherwise
+ */
 int		cub3d_init_render(t_data *data);
+/**
+ * @brief Main function to render the game.
+ * data->player.fov * 0.5f is used to calculate the plane for the camera
+ *
+ * @param[in,out] data data structure
+ */
 void	cub_render(t_data *data);
-
+/**
+ * @brief Main loop of rendering, updates player position, 
+ * map and minimap rendering
+ * 
+ * @param[in,out] data data structure
+ */
 void	cub_loop(t_data *data);
+/**
+ * @brief Update the player position according to inputs
+ * 
+ * @param[in,out] data data structure
+ */
 void	cub_player_update(t_data *data);
+/**
+ * @brief Render the minimap
+ * 
+ * @param[in,out] data data structure
+ */
 void	cub_render_minimap(t_data *data);
+/**
+ * @brief Function to draw a column for rendering
+ * 
+ * @param[in,out] data data structure
+ * @param[in] x x axis column
+ * @param[in,out] rdata structure to handle important values for rendering 
+ */
 void	draw_column(t_data *data, int x, const t_raydata *rdata);
+/**
+ * @brief Check which wall will be hit
+ * 
+ * @param[in,out] data data structure
+ * @param[in,out] rdata structure to handle important values for rendering 
+ */
 void	hitwall_loop(const t_data *data, t_raydata *rdata);
+/**
+ * @brief Function to set the texture based on the hit side and step direction.
+ * 
+ * @param[in] tex pointer to texture structure
+ * @param[in] hitside hitside
+ * @param[in] step step direction
+ */
 t_img	*hitside_texture(t_textures *tex, int hitside, const t_pos2 *step);
+/**
+ * @brief Function to calculate rays using window size and camera angle
```

Comment:
Unless this function has variants, the arguments should convey that rays are calculated based on these 2 variables.